miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -307,21 +310,20 @@ export.set_code
     # => []
 end
 
-#! Applies offset to provided storage slot index for storage access
+#!  Applies storage offset to provided storage slot index for storage access
 #!
 #! Panics:
-#! - slot index is out of bounds
 #! - computed index is out of bounds
 #!
-#! Stack: [storage_offset, slot_index]
+#! Stack: [storage_offset, storage_size, slot_index]
 #! Output: [offset_slot_index]
 export.apply_storage_offset
-    # verify that slot_index is in bounds
-    dup.1 exec.memory::get_num_storage_slots dup movdn.4 lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
-    # => [storage_offset, slot_index, num_storage_slots]
+    # offset index
+    dup movup.3 add
+    # => [offset_slot_index, storage_offset, storage_size]
 
-    # verify that the computed offset_slot_index is in bounds
-    add dup movup.2 lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
+    # verify that slot_index is in bounds
+    movdn.2 add sub.1 dup.1 gte assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
```

Comment:
Why do we need to do `sub.1` here? Could we not skip it and use `gt` instead of `gte`?